sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -350,13 +384,60 @@ class command_graph : public detail::modifiable_command_graph {
 template <>
 class command_graph<graph_state::executable>
     : public detail::executable_command_graph {
-
 protected:
   friend command_graph<graph_state::executable>
   detail::modifiable_command_graph::finalize(const sycl::property_list &) const;
   using detail::executable_command_graph::executable_command_graph;
 };
 
+namespace detail {
+class __SYCL_EXPORT dynamic_parameter_base {
+public:
+  dynamic_parameter_base(
+      sycl::ext::oneapi::experimental::command_graph<graph_state::modifiable>
+          Graph);
+
+  void register_with_node(int ArgIndex, handler &CGH);
```

Comment:
in our spec draft this is just called `register()` rather than `register_with_node()`

Reply:
I hadn't updated this in the spec yet but I ran into conflicts with the `register` keyword in C++ that I don't believe are possible to work around, so I renamed the function instead.

Reply:
hadn't considered that, but a very good point. `register_with_node` sounds fine to me